### PR TITLE
add tests for .devfile.yaml support

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

**What type of PR is this?**
 /kind tests

**What does this PR do / why we need it**:
Add tests for `.devfile.yaml` support
**Which issue(s) this PR fixes**: 

Fixes https://github.com/openshift/odo/issues/5167

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Review by feloy — CHANGES_REQUESTED** [review]

File: tests/integration/devfile/cmd_dot_devfile_test.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-debugrun.yaml"), filepath.Join(commonVar.Context, ".devfile.yaml"))
+			helper.Cmd("odo", "push", "--debug", "--context", projectDir).ShouldPass()
+		})
+		It("should log debug command output", func() {
+			output := helper.Cmd("odo", "log", "--debug", "--context", projectDir).ShouldPass().Out()
+			Expect(output).To(ContainSubstring("ODO_COMMAND_DEBUG"))
+		})
+	})
+
+	Context("Operators are installed in the cluster", func() {
```
[the remark this answers is not in the document]
Comment (anandrkskd) [reply tests/integration/devfile/cmd_dot_devfile_test.go:150]: I think we can remove the above test of `odo service`  and replace it with this one, as it's also testing the `service create` for us in a way.

File: tests/integration/devfile/cmd_dot_devfile_test.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+	AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+
+	When("Creating a nodejs component and replace devfile.yaml to .devfile.yaml", func() {
+		var _ = BeforeEach(func() {
+			helper.Cmd("odo", "create", "--project", commonVar.Project, cmpName, "--devfile", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile.yaml")).ShouldPass()
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, ".devfile.yaml"))
```
[the remark this answers is not in the document]
Comment (anandrkskd) [reply tests/integration/devfile/cmd_dot_devfile_test.go:42]: But is it not the devfile that we are using for testing, I think it's better to use the devfile we have on our example, and if I remember correctly we faced protocol error some time before.

File: tests/integration/devfile/cmd_dot_devfile_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (anandrkskd) [reply tests/integration/devfile/cmd_dot_devfile_test.go:150]: As discussed in the call we want to merge but the devfile we are using for this test is different then the devfile that we are using in the test in the starting of the file. I think we can use this context and move the test which is in the starting of file to this context

File: tests/integration/devfile/cmd_dot_devfile_test.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+				})
+
+				It("should verify if url is created and pushed", func() {
+					stdout = helper.Cmd("odo", "url", "list").ShouldPass().Out()
+					helper.MatchAllInOutput(stdout, []string{url1, "Locally Deleted", "true", "ingress"})
+				})
+			})
+		})
+
+		When("should list the storage with the proper states and container names", func() {
```
Comment (feloy) [inline tests/integration/devfile/cmd_dot_devfile_test.go:72]:
```suggestion
		When("creating a storage", func() {
```

File: tests/integration/devfile/cmd_dot_devfile_test.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+	BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+		cmpName = helper.RandString(6)
+		helper.Chdir(commonVar.Context)
+
+		odoArgs := []string{"catalog", "list", "services"}
+		operators := []string{"redis-operator"}
+		for _, operator := range operators {
+			helper.WaitForCmdOut("odo", odoArgs, 5, true, func(output string) bool {
```
Comment (feloy) [inline tests/integration/devfile/cmd_dot_devfile_test.go:28]: This operation can be quite long, please place this block in a BeforeEach where it is needed only, and not the more englobing one (in `When("Creating a nodejs component and replace devfile.yaml with .devfile.yaml"` I think).

**Review by feloy — CHANGES_REQUESTED** [review]

**Pull request merged**